modules/imaging_uploader/jsx/UploadForm.js

```diff
@@ -195,17 +195,15 @@ class UploadForm extends React.Component {
         });
       },
       error: function(err) {
-        const errMessage = "The following errors occured while " +
-          "attempting to display this page:";
-        let responseText = err.responseText;
-        if (responseText.indexOf(errMessage) > -1) {
-          responseText = responseText.replace('history.back()', 'location.reload()');
-          document.open();
-          document.write(responseText);
-          document.close();
-        }
-        console.error(err);
-      }
+          console.log(err);
+          swal({
+            title: "Errors with Field Values",
+            text: "Display errors here",
+            type: "error",
+            confirmButtonTexr: 'OK'
+          });
+          this.setState({uploadProgress: -1});
+       }.bind(this),
```

Comment:
@davidblader should we be encouraging the use of arrow functions over the the binding of `this`?

Reply:
not sure. both ways are perfectly fine. i guess it would boil down to: what's more confusing, `.bind(this)` or `() => {...}`?
which, in my opinion, are probably about equally confusing for someone unfamiliar with JS, but i would have to go with `.bind(this)` since at least that is consistent with the rest of the code base, but i don't have a strong enough opinion one way or the other